### PR TITLE
Allow spaces in image url

### PR DESCRIPTION
**Pull request by facelessuser**
Preserving general pre-existing title output, ensure that image URL can have spaces (ref #590).

### CODE REVIEW

**Comment by facelessuser** [comment]
Failing issues should be ignored as this issue is not introducing any new failures, but the flake8 issues are being addressed here: https://github.com/Python-Markdown/markdown/pull/593.

**Comment by waylan** [comment]
Interesting. I didn't remember that we actually had tests confirming the preexisting behavior. I'm curious, do we have tests confirming that spaces work in links? If not, then perhaps we're making the wrong change here.

**Comment by facelessuser** [comment]
> I'm curious, do we have tests confirming that spaces work in links?

I don't know yet, but more importantly, we don't have any checks to ensure that they *don't* work or it would already be failing. Omission is some times as bad as explicitly doing something wrong.

The problem is that (for who knows how long) we've allowed them, and possibly given the expectation that this is something that Python Markdown supports.  I'm not sure how many documents we might break.

I'm not sure the motivation against specifically checking that it doesn't support it in images and not links was.

Anyways, the work is done, if we decide to not accept it that is fine.  I think we should either explicitly disallow this in the future, or or go ahead and relax image tests as done here.  Again we may break a good number of document URLs explicitly asserting on this situation, but if we are okay with that, then we should do that instead.

**Comment by waylan** [comment]
My thinking is that if we have tests explicitly requiring that spaces in links be supported, then we should go forward with this change. However, if no such tests exist, my assumption is that the support for it was accidental and we should not go forward with this change. Either way, I agree that changing the link behavior could break too many existing documents so we won't do that, even if that is my preferred way forward.

**Comment by facelessuser** [comment]
If we don't add a test to fail on it, how do we ensure it doesn't get accidentally changed in the future? Or is that not a concern?

Either way, I'll take a look at allowed links.

**Comment by facelessuser** [comment]
I guess we could add tests to ensure it keeps working without updating image links.

**Comment by waylan** [comment]
> If we don't add a test to fail on it, how do we ensure it doesn't get accidentally changed in the future?

What? I think we're talking about different things. When I say "links" I am not referring to images. If links (eg: `[foo](http://example.con/foo bar.html "title")`) already have tests with spaces then we should go forward with this change.  However, if no such tests exist, we should not go forward with this change in image behavior.

Whether we add tests for links with spaces is another matter  which I have not commented on.

**Comment by facelessuser** [comment]
Understood, I was thinking about both images and non images. I'm just looking for a way to make sure we're handling both in a sane manner.  We're on the same page.

**Comment by waylan** [comment]
> I'm just looking for a way to make sure we're handling both in a sane manner. 

To me, being "right" is more important that being consistent in this instance. If the current link behavior was an accident (based on the fact that there are no tests to support it), then I'm saying that we should not change images to match. I'd rather have the inconsistency (for the previously stated backward compatibility reasons) than change the behavior of images. However, if there are supporting tests for the current link behavior, then that was intentional and we should change images to be consistent. 

In other words, before I accept this, someone needs to look through the existing tests and point out any tests which enforce support for spaces in links. If no such tests can be found, then I'm inclined to close instead.

**Comment by facelessuser** [comment]
Understood. That's why I said sane instead of consistent. I just want to make sure whatever we decide on is preserved, but yes these are two separate things. I was just being unclear on what I was talking about, so I'll keep the topic on images.

**Comment by facelessuser** [comment]
I don't see any tests that ensure link spaces work.  Even commonmark doesn't support spaces in their links 🙂 .  I guess the question now is what to do with with the normal links.

**Comment by facelessuser** [comment]
I'm going to close this since there is no actual tests that ensure spaces are allowed for normal links, it makes no sense to add support for spaces in image links.
